### PR TITLE
Merge dev/merge-file-manager-to-dev-0.1.3 into development/v0.1.3

### PR DESCRIPTION
**Pull request by jiayingy**
Implemented on FileManager
- Add new File/Folder
- Delete File/Folder
- Rename File/Folder
- Download File (currently does not support downloading of folder)
- Searching of file and folder (done by @amoshydra)

Wrote test cases for:
**Editor Viewer**
- Toggling of editor, viewer and split screen view

**File Manager**
- Creating of file/folder
- Deleting of file/folder
- Focusing on file/folder will enable delete and download button
- Toggling of icon and list view of documents

### CODE REVIEW

File: sashimi-webapp/src/components/file-manager/FileManager.vue [hunk 1]
```diff
@@ -16,10 +24,102 @@ export default {
...
+      } else {
+        this.$router.push({ path: '', query: { folder: newFolder.id } });
+      }
+      this.docs = newFolder;
+      this.history.update(this.docs);
+    },
+    changeViewMode(viewMode) {
+      this.viewMode = viewMode;
+    },
+    executeAction(action, doc) {
```
Comment (amoshydra) [inline sashimi-webapp/src/components/file-manager/FileManager.vue:47]: Definitely need to refactor this part next time by grouping the relevant conditions together.

e.g. [Creation]. [History], [Operation] etc

**Review by tayjiehao — COMMENTED** [review]
Looks good.

File: sashimi-webapp/src/components/file-manager/FileManager.vue [hunk 2]
```diff
@@ -16,10 +24,102 @@ export default {
...
+      const fileName = doc.name;
+      element.setAttribute('download', fileName);
+
+      element.style.display = 'none';
+      document.body.appendChild(element);
+
+      element.click();
+
+      document.body.removeChild(element);
+    }
```
Comment (amoshydra) [inline sashimi-webapp/src/components/file-manager/FileManager.vue]: Perhaps, use a different name for the "downloadDoc, download"

**Review by chuajiaxuan — APPROVED** [review]
Can add some comments for readability, but looks good! Can proceed to merge!

File: sashimi-webapp/src/components/file-manager/FileManager.vue (hunk 1 above)
Comment (amoshydra) [reply sashimi-webapp/src/components/file-manager/FileManager.vue:47]: https://github.com/nus-mtp/sashimi-note/issues/303

**Pull request merged**